docs/generator_template.md

```diff
@@ -0,0 +1,569 @@
+
+
+    def sendTemperatureChange(self, id):
+      # 5
+        topic = "temperature/changed"
+        # 6 
+        self.client.publish(topic, id)
+```
+
+Make sure you have the paho-mqtt library installed. You can install it using pip with the `pip install paho-mqtt` command.
+Let's break down the previous code snippet:
+
+1. Imports the MQTT module from the paho package, which provides the MQTT client functionality.
+2. Assigns the MQTT broker address `test.mosquitto.org` to the variable MQTT broker. This specifies the location where the MQTT client will connect to.
+3. Creates an instance of the MQTT client object. This object will be used to establish a connection with the MQTT broker and perform MQTT operations.
+4. Establishes a connection to the MQTT broker. This command connects the MQTT client to the broker.
+5. The MQTT topic to which the client will publish messages.
+6. This command publishes the temperature change information to the MQTT broker.
+
+In summary, this code sets up an MQTT client using the paho-MQTT library. It connects to the `test.mosquitto.org` MQTT broker, and the `sendTemperatureChange()` method publishes temperature change information to the `temperature/changed` topic whenever called.
+
+### 2.Test the client
+
+You'll interact with the Temperature Service using the client module you created above. You'll create an instance of the client using `client = TemperatureServiceClient()` and then use `client.sendTemperatureChange` function to publish messages that Temperature Service is subscribed to.
+Create a **test/project/test.py** file in your project and add the code snippet below:
+
+``` python
+from client import TemperatureServiceClient
+from random import randrange
+import time
+
+client = TemperatureServiceClient()
+
+id_length = 8
+min_value = 10**(id_length-1)  # Minimum value with 8 digits (e.g., 10000000)
+max_value = 10**id_length - 1  # Maximum value with 8 digits (e.g., 99999999)
+
+while True:
+    randomId = randrange(min_value, max_value + 1)
+    client.sendTemperatureChange(randomId)
+    print("New temperature detected " + str(randomId) + " sent to temperature/changed")
+    time.sleep(1)
+
+```
+
+Run the code above on your terminal using the command `python test.py`. You'll should see output similar to the snippet below logged on your terminal:
+
+``` cmd
+New temperature detected 64250266 sent to temperature/changed
+New temperature detected 36947728 sent to temperature/changed
+New temperature detected 72955029 sent to temperature/changed
+```
+
+To make sure your client works, also test it using an [MQTT CLI](https://hivemq.github.io/mqtt-cli/) using docker. Run the command `docker run hivemq/mqtt-cli sub -t temperature/changed -h test.mosquitto.org` on your teminal. It will download the image if you don't have it locally then the CLI will conncet to the broker, subscribe to the `temperature/changed` topic and then output the temperature ids on the terminal.
```

Comment:
```suggestion
To make sure your `test.py` and client code works check if the broker really receives temperature-related messages. You can do it using an [MQTT CLI](https://hivemq.github.io/mqtt-cli/) using docker. Run the command `docker run hivemq/mqtt-cli sub -t temperature/changed -h test.mosquitto.org` in your terminal. It will download the image if you don't have it locally, then the CLI will connect to the broker, subscribe to the `temperature/changed` topic and then output the temperature ids on the terminal.
```

also please like in case of test.py terminal output, also show MQTT CLI client output